courseware/views_test.py

```diff
@@ -0,0 +1,12 @@
+"""Test courseware views"""
+import pytest
+from django.shortcuts import reverse
+from rest_framework import status
+
+pytestmark = [pytest.mark.django_db]
```

Comment:
This doesn't need the `django_db` mark, does it?

Reply:
Surprisingly it does:

```
______________________________________________ test_index_view ______________________________________________

client = <django.test.client.Client object at 0x7f8ebc6a15c0>

    def test_index_view(client):
        """Verify the index view is as expected"""
>       response = client.get(reverse("mitxpro-index"))

mitxpro/views_test.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.6/site-packages/django/test/client.py:527: in get
    response = super().get(path, data=data, secure=secure, **extra)
/usr/local/lib/python3.6/site-packages/django/test/client.py:339: in get
    **extra,
/usr/local/lib/python3.6/site-packages/django/test/client.py:414: in generic
    return self.request(**r)
/usr/local/lib/python3.6/site-packages/django/test/client.py:477: in request
    response = self.handler(environ)
/usr/local/lib/python3.6/site-packages/django/test/client.py:140: in __call__
    response = self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py:78: in get_response
    response = self._middleware_chain(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:90: in __call__
    response = self.process_request(request)
/usr/local/lib/python3.6/site-packages/wagtail/core/middleware.py:13: in process_request
    request.site = Site.find_for_request(request)
/usr/local/lib/python3.6/site-packages/wagtail/core/models.py:120: in find_for_request
    return get_site_for_hostname(hostname, port)
/usr/local/lib/python3.6/site-packages/wagtail/core/sites.py:35: in get_site_for_hostname
    'root_page'
/usr/local/lib/python3.6/site-packages/django/db/models/query.py:268: in __iter__
    self._fetch_all()
/usr/local/lib/python3.6/site-packages/nplusone/ext/django/patch.py:295: in fetch_all
    original_fetch_all(self)
/usr/local/lib/python3.6/site-packages/django/db/models/query.py:1186: in _fetch_all
    self._result_cache = list(self._iterable_class(self))
/usr/local/lib/python3.6/site-packages/django/db/models/query.py:54: in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
/usr/local/lib/python3.6/site-packages/django/db/models/sql/compiler.py:1063: in execute_sql
    cursor = self.connection.cursor()
/usr/local/lib/python3.6/site-packages/django/db/backends/base/base.py:255: in cursor
    return self._cursor()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.backends.postgresql.base.DatabaseWrapper object at 0x7f8ecf03ad68>, name = None

    def _cursor(self, name=None):
>       self.ensure_connection()
E       Failed: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it.

/usr/local/lib/python3.6/site-packages/django/db/backends/base/base.py:232: Failed
_____________________________________________ test_webpack_url ______________________________________________

mocker = <pytest_mock.MockFixture object at 0x7f8ebc7ae7b8>
settings = <pytest_django.fixtures.SettingsWrapper object at 0x7f8ebc92f1d0>
client = <django.test.client.Client object at 0x7f8ebc92fc88>

    def test_webpack_url(mocker, settings, client):
        """Verify that webpack bundle src shows up in production"""
        settings.GA_TRACKING_ID = "fake"
        settings.ENVIRONMENT = "test"
        settings.VERSION = "4.5.6"
        settings.USE_WEBPACK_DEV_SERVER = False
        get_bundle = mocker.patch("mitxpro.templatetags.render_bundle._get_bundle")
    
>       response = client.get(reverse("login"))

mitxpro/views_test.py:27: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.6/site-packages/django/test/client.py:527: in get
    response = super().get(path, data=data, secure=secure, **extra)
/usr/local/lib/python3.6/site-packages/django/test/client.py:339: in get
    **extra,
/usr/local/lib/python3.6/site-packages/django/test/client.py:414: in generic
    return self.request(**r)
/usr/local/lib/python3.6/site-packages/django/test/client.py:477: in request
    response = self.handler(environ)
/usr/local/lib/python3.6/site-packages/django/test/client.py:140: in __call__
    response = self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py:78: in get_response
    response = self._middleware_chain(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:91: in __call__
    response = response or self.get_response(request)
/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
/usr/local/lib/python3.6/site-packages/django/utils/deprecation.py:90: in __call__
    response = self.process_request(request)
/usr/local/lib/python3.6/site-packages/wagtail/core/middleware.py:13: in process_request
    request.site = Site.find_for_request(request)
/usr/local/lib/python3.6/site-packages/wagtail/core/models.py:120: in find_for_request
    return get_site_for_hostname(hostname, port)
/usr/local/lib/python3.6/site-packages/wagtail/core/sites.py:35: in get_site_for_hostname
    'root_page'
/usr/local/lib/python3.6/site-packages/django/db/models/query.py:268: in __iter__
    self._fetch_all()
/usr/local/lib/python3.6/site-packages/nplusone/ext/django/patch.py:295: in fetch_all
    original_fetch_all(self)
/usr/local/lib/python3.6/site-packages/django/db/models/query.py:1186: in _fetch_all
    self._result_cache = list(self._iterable_class(self))
/usr/local/lib/python3.6/site-packages/django/db/models/query.py:54: in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
/usr/local/lib/python3.6/site-packages/django/db/models/sql/compiler.py:1063: in execute_sql
    cursor = self.connection.cursor()
/usr/local/lib/python3.6/site-packages/django/db/backends/base/base.py:255: in cursor
    return self._cursor()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.backends.postgresql.base.DatabaseWrapper object at 0x7f8ecf03ad68>, name = None

    def _cursor(self, name=None):
>       self.ensure_connection()
E       Failed: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it.

/usr/local/lib/python3.6/site-packages/django/db/backends/base/base.py:232: Failed
```